src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/ListenerPrimary.cs

```diff
@@ -179,5 +184,53 @@ private struct FILE_COMPLETION_INFORMATION
                 }, this).ConfigureAwait(false);
             }
         }
+
+        private class PipeReadContext
+        {
+            private readonly ListenerPrimary _listener;
+            private readonly IntPtr _bufPtr;
+            private int _bytesRead;
+
+            public PipeReadContext(ListenerPrimary listener)
+            {
+                _listener = listener;
+                _bufPtr = Marshal.AllocHGlobal(8);
+            }
+
+            public Libuv.uv_buf_t AllocCallback(UvStreamHandle dispatchPipe, int suggestedSize)
+            {
+                return dispatchPipe.Libuv.buf_init(_bufPtr + _bytesRead, 8 - _bytesRead);
+            }
+
+            public void ReadCallback(UvStreamHandle dispatchPipe, int status)
+            {
+                try
+                {
+                    dispatchPipe.Libuv.ThrowIfErrored(status);
+
+                    _bytesRead += status;
```

Comment:
YAGNI. Never experienced a prolonged pipe connection that wasn't our own. We're fixing a pretty specific issue.